CI: Lint image build logic / Check Cilium Envoy image fails on main with "images: No such file or directory"

### Test Name

Lint image build logic / Check Cilium Envoy image

### Test Failure Output

```shell
### Check Cilium Envoy image


Run make -C images check-envoy-image
make: *** images: No such file or directory.  Stop.
Error: Process completed with exit code 2.


## Merge Upload and Status / Fail job if result is failure

### Fail job if result is failure


Run echo "::error::Workflow failed because one or more jobs failed"
Error: Workflow failed because one or more jobs failed
Error: Process completed with exit code 1.


Additional affected runs with same error:
- https://github.com/cilium/cilium/actions/runs/27543112302 (renovate/main-goswagger)
- https://github.com/cilium/cilium/actions/runs/27542994042 (main)
```

### CI Build Link

https://github.com/cilium/cilium/actions/runs/27544422518

### Relevant Log Output

```shell

```

### Sysdump / Zip File

_No response_

--- a/.github/workflows/lint-images-base.yaml
+++ b/.github/workflows/lint-images-base.yaml
@@ -103,7 +103,7 @@ jobs:
           args: -C untrusted/images check-runtime-image RUNTIME_IMAGE=${{ env.CILIUM_RUNTIME_IMAGE }}
 
       - name: Check Cilium Envoy image
-        run: make -C images check-envoy-image
+        run: make -C trusted/images check-envoy-image
 
   merge-upload-and-status:
     name: Merge Upload and Status

